jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+import java.util.Collection;
+import javax.enterprise.event.Event;
+
+import org.jbpm.workbench.ks.integration.event.ServerInstanceRegistered;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.kie.api.KieServices;
+import org.kie.api.runtime.KieContainer;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.balancer.LoadBalancer;
+import org.kie.server.client.impl.AbstractKieServicesClientImpl;
+import org.kie.server.controller.api.model.events.ServerInstanceConnected;
+import org.kie.server.controller.api.model.events.ServerInstanceDisconnected;
+import org.kie.server.controller.api.model.events.ServerTemplateDeleted;
+import org.kie.server.controller.api.model.runtime.ServerInstance;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.workbench.common.screens.server.management.service.SpecManagementService;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(Parameterized.class)
+public class KieServerIntegrationTest {
+
+    @Parameterized.Parameter(0)
+    public String serverTemplateId1;
+
+    @Parameterized.Parameter(1)
+    public String serverTemplateId2;
+
+    @Mock
+    SpecManagementService specManagementService;
+
+    @Mock
+    KieServices kieServices;
+
+    @Mock
+    Event<ServerInstanceRegistered> serverInstanceRegisteredEvent;
+
+    @InjectMocks
+    KieServerIntegration kieServerIntegration;
+
+    private static ContainerSpec newContainerSpec() {
+        final ContainerSpec spec = new ContainerSpec();
+        spec.setId("id");
+        spec.setContainerName("name");
+        spec.setStatus(KieContainerStatus.STARTED);
+        spec.setReleasedId(new ReleaseId("groupId",
+                                         "artifactId",
+                                         "1.0"));
+        return spec;
+    }
+
+    @Parameterized.Parameters(name = "Server Template 1: {0}, Server Template 2: {1}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {"test-kie-server", "test-kie-server2"},
+                {"test-kie-server2", "test-kie-server"},
+                {"test-kie-server1", "test-kie-server2"},
+                {"test-kie-server2", "test-kie-server1"},
+                {"test-kie-server|", "test-kie-server"},
+                {"test-kie-server", "test-kie-server|"},
+        });
+    }
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        when(kieServices.newKieContainer(any())).thenReturn(mock(KieContainer.class));
+    }
+
+    @Test
+    public void testSimilarKieServerName() {
```

Comment:
I was thinking suggesting adding one more test which covers the case when two instances use the same server template, but I'm not sure how to fit that into this parameterized approach :)

Reply:
@bsremac we can always add more test classes :)